tests/base/unsubscribe_subscriptor.cc

```diff
@@ -38,9 +38,11 @@ main()
 
   Subscriptor       subscriptor;
   std::atomic<bool> dummy_a;
-  subscriptor.subscribe(&dummy_a, "a");
-  subscriptor.unsubscribe(&dummy_a, "b");
-  subscriptor.unsubscribe(&dummy_a, "a");
+  const char *      foo_a = "a";
+  const char *      foo_b = "b";
+  subscriptor.subscribe(&dummy_a, foo_a);
```

Comment:
oh, that's unfortunate. are you saying that you can't just pass `"a"` any more? What do you do in a context where you can't create a named variable -- say, in a member initializer list?

Reply:
That was never safe to do according to the standard. It is implementation-defined if these string literals are stored at the same memory address or not.

An object should store this string in some variable and pass the variable just as `SmartPointer` does. I am not sure how you would call `subscribe` in an initializer list, but that should work as well if there is a variable storing the string.